--- uTox-0.17.2/CMakeLists.txt	2020-05-11 21:11:12.467889017 +0500
+++ uTox-0.17.2/CMakeLists.txt	2020-05-11 21:14:55.907888165 +0500
@@ -318,12 +318,12 @@
 endif()
 
 # Acquire 3rd party headers
-find_path(STBI_INCLUDE_DIR NAMES stb_image.h stb_image_write.h PATHS third-party/stb)
+find_path(STBI_INCLUDE_DIR NAMES stb_image.h stb_image_write.h PATHS /usr/include/stb)
 if((NOT STBI_INCLUDE_DIR) OR (NOT EXISTS ${STBI_INCLUDE_DIR}))
     message("STBI not found")
-    execute_process(COMMAND git submodule update --init -- third-party/stb WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+    #execute_process(COMMAND git submodule update --init -- third-party/stb WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
     # set FOO_INCLUDE_DIR properly
-    set(STBI_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third-party/stb/ CACHE PATH "stbi include directory")
+    set(STBI_INCLUDE_DIR /usr/include/stb/ CACHE PATH "stbi include directory")
 endif()
 
 
--- uTox-0.17.2/src/stb.c	2020-04-13 04:46:19.000000000 +0500
+++ uTox-0.17.2/src/stb.c	2020-05-11 21:19:01.606887227 +0500
@@ -1,4 +1,4 @@
 #define STB_IMAGE_IMPLEMENTATION
-#include "../third-party/stb/stb_image.h"
+#include <stb/stb_image.h>
 #define STB_IMAGE_WRITE_IMPLEMENTATION
-#include "../third-party/stb/stb_image_write.h"
+#include <stb/stb_image_write.h>
--- uTox-0.17.2/src/stb.h	2020-04-13 04:46:19.000000000 +0500
+++ uTox-0.17.2/src/stb.h	2020-05-11 21:20:24.141886913 +0500
@@ -1,8 +1,8 @@
 #ifndef STB_H
 #define STB_H
 
-#include "../third-party/stb/stb_image.h"
-#include "../third-party/stb/stb_image_write.h"
+#include <stb/stb_image.h>
+#include <stb/stb_image_write.h>
 
 // uTox can't find this function unless declared here.
 extern unsigned char *stbi_write_png_to_mem(unsigned char *pixels, int stride_bytes,
